dev/io.openliberty.cdi.4.0.internal_fat/fat/src/io/openliberty/cdi40/internal/fat/bce/basicwar/permissions.xml

```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<permissions xmlns="https://jakarta.ee/xml/ns/jakartaee"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="https://jakarta.ee/xml/ns/jakartaee
+        https://jakarta.ee/xml/ns/jakartaee/permissions_9.xsd"
+    version="9">
+    <!-- Update to version=10 when implemented -->
+    <permission>
+        <class-name>java.lang.RuntimePermission</class-name>
+        <name>accessDeclaredMembers</name>
```

Comment:
Which bit of application code needs this permission?

Reply:
Ah, this should have a comment.

The application code doesn't need it, but it uses the CDI SPI, which allows access to declared members.

Ideally, CDI would declare which permissions are needed by which methods, or define and check its own permission but currently it doesn't. With `SecurityManager` deprecated in Java 17, I think it's unlikely that this will change.

Weld _could_ wrap everything in `doPriviliged` calls, but that would completely subvert these runtime security checks by allowing access to the same information by a different API, so I think it's better not to do that.